worlds/animal_well/client.py

```diff
@@ -170,6 +215,60 @@ def on_package(self, cmd: str, args: dict):
         if cmd == "Connected":
             self.slot_data = args.get("slot_data", {})
 
+        try:
+            def parseIds(data):
+                for item in data:
+                    if 'type' in item and 'text' in item:
+                        if item['type'] == 'player_id':
+                            item['text'] = self.player_names[int(item['text'])]
+                        elif item['type'] == 'item_id':
+                            item['text'] = item_id_to_name[int(item['text'])]
+                        elif item['type'] == 'location_id':
+                            item['text'] = location_id_to_name[int(item['text'])]
+
+                return data
+
+            if cmd == "PrintJSON":
+                msgType = args.get("type")
+                if msgType == "Chat":
+                    display_to_client(self, "".join(o['text'] for o in parseIds(args.get("data"))))
+                elif msgType == "Hint":
+                    text = "".join(o['text'] for o in parseIds(args.get("data")))
+                    display_to_client(self, text)
+                elif msgType == "Join":
+                    display_to_client(self, args.get('data')[0]['text'])
+                elif msgType == "CommandResult":
+                    display_to_client(self, args.get('data')[0]['text'])
+                elif msgType == "Tutorial":
+                    pass
+                elif msgType == "ItemSend":
+                    data = parseIds(args.get('data'))
+                    text = "".join(o['text'] for o in data).replace(')', '').replace('(', 'at ')
+                    display_to_client(self, text)
```

Comment:
```suggestion
                elif msgType == "ItemSend":
                    destination_player_id = args["receiving"]
                    source_player_id = args["item"][2]  # it's a tuple, so we can't index by name
                    self_slot: int = self.slot
                    # we don't want to display every item message, just ones relevant to us
                    if self_slot not in [source_player_id, destination_player_id]:
                        return
                    item_id = args["item"][0]
                    location_id = args["item"][1]
                    # classification = args["item"][3]  # may use later if we can color-code them in-game
                    item_name = self.item_names.lookup_in_slot(item_id, destination_player_id)
                    location_name = self.location_names.lookup_in_slot(location_id, source_player_id)
                    text = "Error occurred, report to ANIMAL WELL AP devs"
                    if self_slot == source_player_id:
                        destination_player_name = self.player_names[destination_player_id]
                        text = f"You sent {item_name} to {destination_player_name} from {location_name}!"
                    if self_slot == destination_player_id:
                        source_player_name = self.player_names[source_player_id]
                        text = f"{source_player_name} sent you your {item_name}!"
                    if self_slot == source_player_id and self_slot == destination_player_id:
                        text = f"You found your {item_name} at {location_name}!"
                    display_to_client(self, text)
```

While looking through the parseIds thing, I ended up just rewriting this altogether. Maybe a little wordy, but it does successfully get the job done, and filter out irrelevant items/locations from showing up in-game (since who cares if someone else sent something to someone else anyway?)

Reply:
Much better than what I'd taped together. I was trying to find where we had a thing in place to get the names from ids, didn't find `lookup_in_slot` somehow, which is why those new list comprehensions got added unnecessarily. 😅